Fix environment variable persistence

DannyBen opened the pull request:
Closes #60 

I added test in the most appropriate place I could think of and verified the test fails without the fix.
All other specs pass locally.

DannyBen commented on spec/unit/run_spec.rb:
```diff
@@ -160,4 +160,15 @@
+    output = StringIO.new
+    command = TTY::Command.new(output: output)
+
+    out, _err = command.run("env | grep FOO", env: {foo: 1})
```
In regards to these style issues - I agree, but followed the style already present in the code and specs

piotrmurach commented on lib/tty/command/cmd.rb:
```diff
@@ -64,7 +64,7 @@ def initialize(env_or_cmd, *args)
       # @api public
       def update(options)
-        @options.update(options.update(@options))
+        @options.update(options.merge(@options))
```
Great catch! Let's raise a glass to state mutation.

piotrmurach commented on spec/unit/run_spec.rb:
```diff
@@ -160,4 +160,15 @@
     ])
   end
+
+  it "does not persist environment variables" do
+    output = StringIO.new
+    command = TTY::Command.new(output: output)
+
+    out, _err = command.run("env | grep FOO", env: {foo: 1})
+    expect(out.chomp).to eq("FOO=1")
+    out, _err = command.run!("env | grep FOO")
+    expect(out).to eq("")
+  end
```
I'd consider examining the `output` format as it will show the commands that are run with the `export` statement. Also, it would be good to get this running on Windows which probably won't have `grep` utility. I tend to stay with the safer `echo` command.

DannyBen replied:
Sure, updating.

DannyBen replied:
I updated the tests, borrowing from a similar test in the same spec.

If it was my code, I would have probably opted to use something like:

```ruby
expect(output.string).to include "..."
```
instead of all the `readlines` acrobatics, but since your gems need to work on more environments, I figured I play it safe since you must have a reason for doing it this way.

---

As a side note, the styling bot is super annoying - if it was a real robot, I would have disassembled it and sold it for parts... 😏

piotrmurach replied:
I'm ok with any approach that verifies behaviour and passes tests. I updated all the test setup to work on GitHub Actions and Appveyor so you should get immediate feedback. The style bot is annoying but it has good intentions...😉

piotrmurach reviewed (commented):
Thank you for contributing this patch so fast! Before I merge, could you please see if you can resolve my comment about the test? Don't worry about the style issue I will look into making them consistent.

piotrmurach reviewed (commented):
Sorry to be a pain and promise this to be the last thing before a merge. Would you have time to add a changelog entry(no need for the release version, only fix entry) and if possible update code style issues? I went through specs and updated them today to reflect the rubocop settings so if you could pull in the changes and rebase that would be great. Thank you.

DannyBen commented:
> Would you have time to...

Yes. Stand by :)

DannyBen commented:
- Pulled and merged upstream master
- Added changelog entry
- Resolved all but one Hound complaints
- Simplified the spec to compare only to the part of the string we care about
